cluster-operator/src/test/java/io/strimzi/operator/cluster/model/cruisecontrol/CruiseControlConfigurationTest.java

```diff
@@ -0,0 +1,213 @@
+            NETWORK_INBOUND_CAPACITY_GOAL.toString(),
+            NETWORK_OUTBOUND_CAPACITY_GOAL.toString(),
+            CPU_CAPACITY_GOAL.toString(),
+            POTENTIAL_NETWORK_OUTAGE_GOAL.toString(),
+            NETWORK_INBOUND_USAGE_DISTRIBUTION_GOAL.toString(),
+            NETWORK_OUTBOUND_USAGE_DISTRIBUTION_GOAL.toString(),
+            CPU_USAGE_DISTRIBUTION_GOAL.toString(),
+            LEADER_BYTES_IN_DISTRIBUTION_GOAL.toString());
+
+    private static KafkaSpec createKafkaSpec(BrokerCapacity brokerCapacity) {
+        return new KafkaBuilder()
+                .withNewSpec()
+                    .withNewCruiseControl()
+                        .withBrokerCapacity(brokerCapacity)
+                    .endCruiseControl()
+                .endSpec()
+                .build()
+                .getSpec();
+    }
+
+    private static BrokerCapacity createBrokerCapacity(String cpuCapacity,
+                                                       String inboundNetworkCapacity,
+                                                       String outboundNetworkCapacity) {
+        return new BrokerCapacityBuilder()
+                .withCpu(cpuCapacity)
+                .withInboundNetwork(inboundNetworkCapacity)
+                .withOutboundNetwork(outboundNetworkCapacity)
+                .build();
+    }
+
+    private static BrokerCapacityOverride createBrokerCapacityOverride(List<Integer> brokerIds,
+                                                                       String cpuCapacity,
+                                                                       String inboundNetworkCapacity,
+                                                                       String outboundNetworkCapacity) {
+        return new BrokerCapacityOverrideBuilder()
+                .addAllToBrokers(brokerIds)
+                .withCpu(cpuCapacity)
+                .withInboundNetwork(inboundNetworkCapacity)
+                .withOutboundNetwork(outboundNetworkCapacity)
+                .build();
+    }
+
+    private static ResourceRequirements createCpuResourceRequirement(String request, String limit) {
+        return new ResourceRequirementsBuilder()
+                .addToRequests(ResourceType.CPU.value(), new Quantity(request))
+                .addToLimits(ResourceType.CPU.value(), new Quantity(limit))
+                .build();
+    }
+
+    private void assertGoalsPresent(List<String> actual, CruiseControlGoals... expected) {
+        for (CruiseControlGoals goal : expected) {
+            assertThat("Expected goal to be present: " + goal, actual.contains(goal.toString()), is(true));
+        }
+    }
+
+    private void assertGoalsAbsent(List<String> actual, CruiseControlGoals... expected) {
+        for (CruiseControlGoals goal : expected) {
+            assertThat("Expected goal to be absent: " + goal, actual.contains(goal.toString()), is(false));
+        }
+    }
+
+    @Test
+    public void testFilterResourceGoalsWithoutCapacityConfig() {
```

Comment:
Do we want to separate out these tests into separate test cases with descriptive method names so that each test case is testing one scenario? Might make it easier to spot where the failure is and debug.